### ISSUE TITLE
Incorrect mapbox-gl peer dependency range (npm7 workspaces)

### ISSUE THREAD

**Issue by dschnare** [mapbox/mapbox-gl-directions#268]
```json
{
  "peerDependencies": {
    "mapbox-gl": "^0.41.0 <2.0.0"
  }
}
```

This semver range computes to `>=0.41.0 <0.42.0-0 <2.0.0`.
This range can only be satisfied with mapbox-gl versions `>= 0.41.0 < 0.42.0`.

This becomes a problem when using npm 7.x and workspaces.

Suggested fix is to change the semver range for the mapbox-gl peer dependency to `0.41.0 <2.0.0`, this will allow mapbox-gl any version below 2.0.0 and above or equal to 0.41.0 to meet the peer dependency which I believe was the original intention.

**Comment by dschnare** [comment]
After review of my suggested change, I believe the range should be this `>=0.41.0 <2.0.0`.

**Comment by Chingili** [comment]
I have the same issue, let me know if there is a "hacky" solution for the time being

**Comment by sezf-opscentral** [comment]
This wasn't a problem for us until our CI/CD pipeline upgraded to Node 16.15.1 & npm 8.11.0 now we see this error during npm install:
```
npm ERR! While resolving: @mapbox/mapbox-gl-directions@4.1.0
npm ERR! Found: mapbox-gl@1.13.2
npm ERR! node_modules/mapbox-gl
npm ERR!   mapbox-gl@"^1.13.2" from the root project
npm ERR!   peer mapbox-gl@">=0.32.1 <2.0.0" from @mapbox/mapbox-gl-supported@1.5.0
npm ERR!   node_modules/@mapbox/mapbox-gl-supported
npm ERR!     @mapbox/mapbox-gl-supported@"^1.5.0" from mapbox-gl@1.13.2
npm ERR!   1 more (ngx-mapbox-gl)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer mapbox-gl@"^0.41.0 <2.0.0" from @mapbox/mapbox-gl-directions@4.1.0
npm ERR! node_modules/@mapbox/mapbox-gl-directions
npm ERR!   @mapbox/mapbox-gl-directions@"^4.1.0" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: mapbox-gl@0.41.0
npm ERR! node_modules/mapbox-gl
npm ERR!   peer mapbox-gl@"^0.41.0 <2.0.0" from @mapbox/mapbox-gl-directions@4.1.0
npm ERR!   node_modules/@mapbox/mapbox-gl-directions
npm ERR!     @mapbox/mapbox-gl-directions@"^4.1.0" from the root project
```

**Comment by dennisvd** [comment]
[Example on mapbox website](https://docs.mapbox.com/mapbox-gl-js/example/mapbox-gl-directions/) shows mapbox-gl 2.8.2 working together with mapbox-gl-directions 4.1.0 

A work-around solution for when you use NPM version > 7 is to run `npm install` with the `--legacy-peer-deps` flag
It seems that the new NPM version has a stricter policy when it comes to peer dependencies.

**Comment by rogadev** [comment]
I spoke too soon. It installs without the above error tree, but doesn't seem to be there when I attempt to import it.

```js
import MapboxDirections from "@mapbox/mapbox-gl-directions";
```


`Cannot find module '@mapbox/mapbox-gl-directions' or its corresponding type declarations.ts(2307)`

**Comment by Tepepaz** [comment]
Currently, it does install with `--legacy-peer-deps` but then it throws an error when launching the app.

`error - ./node_modules/@mapbox/mapbox-gl-directions/src/controls/inputs.js:6:0
Module not found: Can't resolve 'fs'

Import trace for requested module:
./node_modules/@mapbox/mapbox-gl-directions/src/directions.js
./node_modules/@mapbox/mapbox-gl-directions/src/index.js`

**Comment by kennydurkin** [comment]
A recent update to [package.json](https://github.com/mapbox/mapbox-gl-directions/pull/298/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519) seems to finally address this in the #298 changeset. 

Seems like we're now just waiting for them to publish a new version to NPM? cc @chriswhong

**Comment by collier-watkins** [comment]
@chriswhong Any updates on this issue?

**Comment by snsd0805yoona** [comment]
Any update on this? @rogadev How do you solve this?

**Comment by Vlad-Vasinev** [comment]
Yep, i also faced with this problem "Cannot find module '@mapbox/mapbox-gl-directions'", any suggestions?

**Comment by makosdanii** [comment]
@chriswhong Could you please tell me if there's a solution, or update upcoming? I would use the navigation package for a school project and the deadline is coming :D

**Comment by gerardo-rodriguez** [comment]
Hello, I'm curious about updates on this issue as well. Thank you! 🙂
